homeassistant/components/poolsense/sensor.py

```diff
@@ -93,6 +96,6 @@ class PoolSenseSensor(PoolSenseEntity, SensorEntity):
     """Sensor representing poolsense data."""
 
     @property
-    def native_value(self):
+    def native_value(self) -> StateType:
         """State of the sensor."""
-        return self.coordinator.data[self.entity_description.key]
+        return cast(StateType, self.coordinator.data[self.entity_description.key])
```

Comment:
Instead of this cast, could the coordinator data be typed as `dict[str, StateType]`?

Reply:
Yea, this should also resolve your other comment.